libvast/vast/fbs/segment.fbs

```diff
@@ -31,6 +31,16 @@ table Segment {
   events: ulong;
 }
 
+namespace vast.fbs;
+
+union SegmentUnion {
```

Comment:
```suggestion
union VersionedSegment {
```

Reply:
I think this is generally a better name, because it matches the `versioned_segment` field below. This makes it easier to make the connection in the generated C++ code.